### PR TITLE
Fixed #34964 -- Avoided migrations when combining or inverting the order of CheckConstraint conditions.

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
ticket-34964

### CODE REVIEW

File: django/db/models/query_utils.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def deconstruct(self):
     def identity(self):
         path, args, kwargs = self.deconstruct()
         identity = [path, *kwargs.items()]
-        for child in args:
+        for child in sorted(args, key=str):
```
Comment (charettes) [inline django/db/models/query_utils.py:160]: Any thoughts about using `hash` instead?

**Review by charettes — COMMENTED** [review]
As brought up on the ticket my only concerns here are that this changes the identify of all usage of `Q` on only for `CheckConstraint.check` where we're pretty confident that the ordering won't matter. I'm not entirely sure this is the case for `Index.condition` and when dealing with complex index expressions though.

I guess an alternative would be to have `CheckConstraint` _sort_ solely its `.check` instead.

File: django/db/models/query_utils.py (hunk 1 above)
Comment (jacobtylerwalls) [reply django/db/models/query_utils.py:160]: Sure, I could do that by moving the sort operation later, after all `make_hashable()` conversions. 👍

**Comment by jacobtylerwalls** [comment]
Closing per ticket. Thanks y'all for the advice & research!

**Pull request closed without merging**